src/solver/optimisation/Group1.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+#include "ConstraintGroup.h"
+#include "AreaBalance.h"
+#include "FictitiousLoad.h"
+#include "ShortTermStorageLevel.h"
+#include "FlowDissociation.h"
+#include "BindingConstraintHour.h"
+
+class Group1 : public ConstraintGroup
+{
+public:
+    using ConstraintGroup::ConstraintGroup;
+
+    /*TODO Rename this*/
+    void Build() override;
+
+private:
+    std::shared_ptr<AreaBalanceData> GetAreaBalanceData(int pdt, uint32_t pays);
```

Comment:
For all those factory methods of plain data structures, it would be better to not use heap allocation, and use the plain structures instead: we should return a plain `AreaBalanceData` here, not a pointer.